server/keyshare/tasks/tasks.go

```diff
@@ -0,0 +1,157 @@
+package tasks
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/go-errors/errors"
+	_ "github.com/jackc/pgx/stdlib"
+	"github.com/privacybydesign/irmago/server/keyshare"
+)
+
+type taskHandler struct {
+	conf *Configuration
+	db   keyshare.DB
+}
+
+func newHandler(conf *Configuration) (*taskHandler, error) {
+	err := processConfiguration(conf)
+	if err != nil {
+		return nil, err
+	}
+	db, err := sql.Open("pgx", conf.DBConnStr)
+	if err != nil {
+		return nil, err
+	}
+	if err = db.Ping(); err != nil {
+		return nil, errors.Errorf("failed to connect to database: %v", err)
+	}
+
+	task := &taskHandler{db: keyshare.DB{DB: db}, conf: conf}
+	return task, nil
+}
+
+func Do(conf *Configuration) error {
+	task, err := newHandler(conf)
+	if err != nil {
+		return err
+	}
+
+	task.cleanupEmails()
+	task.cleanupTokens()
+	task.cleanupAccounts()
+	task.expireAccounts()
+
+	return nil
+}
+
+// Remove email addresses marked for deletion long enough ago
+func (t *taskHandler) cleanupEmails() {
+	_, err := t.db.Exec("DELETE FROM irma.emails WHERE delete_on < $1", time.Now().Unix())
+	if err != nil {
+		t.conf.Logger.WithField("error", err).Error("Could not remove email addresses marked for deletion")
+	}
+}
+
+// Remove old login and email verifciation tokens
+func (t *taskHandler) cleanupTokens() {
+	_, err := t.db.Exec("DELETE FROM irma.email_login_tokens WHERE expiry < $1", time.Now().Unix())
+	if err != nil {
+		t.conf.Logger.WithField("error", err).Error("Could not remove email login tokens that have expired")
+		return
+	}
+	_, err = t.db.Exec("DELETE FROM irma.email_verification_tokens WHERE expiry < $1", time.Now().Unix())
+	if err != nil {
+		t.conf.Logger.WithField("error", err).Error("Could not remove email verification tokens that have expired")
+	}
+}
+
+// Cleanup accounts disabled long enough ago.
+func (t *taskHandler) cleanupAccounts() {
+	_, err := t.db.Exec("DELETE FROM irma.users WHERE delete_on < $1 AND (coredata IS NULL OR last_seen < delete_on - $2)",
+		time.Now().Unix(),
+		t.conf.DeleteDelay*24*60*60)
+	if err != nil {
+		t.conf.Logger.WithField("error", err).Error("Could not remove accounts scheduled for deletion")
+	}
+}
```

Comment:
Wouldn't it be more neat to reset `delete_on` back to NULL when the logs into his/her IRMA-app to prevent the account from getting inactive. Now `delete_on` keeps always set. That could lead to mistakes in the future if we forget this is the case and we make changes.